### PR TITLE
Add callback to report rendered beat config under elastic-agent

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

CC @cmacknz 

This adds a debug hook to to beats that reports the final "rendered" config to elastic-agent's debug callaback system. When a user runs `elastic-agent diagnostics`, this enables beats to report the config as seen by beats, as opposed to the "raw" config viewed by elastic-agent.

## Why is it important?

This can be useful for debugging issues with configs in beats.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by cmacknz** [comment]
Haven't looked closely yet but I think we need this in 8.6 to be able to debug problems, added the backport label.

File: x-pack/libbeat/management/managerV2.go [hunk 1]
```diff
@@ -378,3 +393,44 @@ func (cm *BeatV2Manager) handleInputReload(unit *client.Unit) {
...
+	// generate output
+	outputCfg := map[string]interface{}{}
+	err := cm.lastOutputCfg.Config.Unpack(&outputCfg)
+	if err != nil {
+		cm.logger.Errorf("error unpacking output config for debug callback: %s", err)
+		return nil
+	}
+	// combine the two in a somewhat coherent way
+	// This isn't perfect, but generating a config that can actually be fed back into the beat
+	// would require
```
Comment (cmacknz) [inline x-pack/libbeat/management/managerV2.go:421]:
```suggestion
	// would require
```

We appear to be missing the rest of this sentence.

**Comment by cmacknz** [comment]
Tried this out, seems to work great! One thing we need to follow up on is getting any credentials that might have been dumped out automatically redacted. We had some code in agent to do this automatically https://github.com/elastic/elastic-agent/pull/566, I'm trying to track down where it went in V2.

File: x-pack/libbeat/management/managerV2.go (hunk 1 above)
Comment (fearful-symmetry) [reply x-pack/libbeat/management/managerV2.go:421]: Yah, looks like something I would do...

**Review by cmacknz — APPROVED** [review]

**Comment by fearful-symmetry** [comment]
@cmacknz should the credential scrubber happen in this PR, or a follow-up?

**Comment by cmacknz** [comment]
> should the credential scrubber happen in this PR, or a follow-up?

I think this needs to happen on the agent side so each individual diagnostic hook in every input process doesn't need to worry about. We can leave that as a follow up.

**Comment by fearful-symmetry** [comment]
E2E errors seem unrelated, merging

**Comment by blakerouse** [comment]
> > should the credential scrubber happen in this PR, or a follow-up?
> 
> I think this needs to happen on the agent side so each individual diagnostic hook in every input process doesn't need to worry about. We can leave that as a follow up.

This assumes that the Elastic Agent knows the results of every configuration that is passed through it. I would agree that Elastic Agent could do a best effort approach but it will always be that, as best effort.

I believe that if the integration knows what should not be included in the debugging output then it should handle the removal before sending it to Elastic Agent. Elastic Agent can apply its own logic after that, but again that would only be best effort.

**Comment by cmacknz** [comment]
Agreed in general, in this case the credential scrubbing for the beat config is exactly the same as is already necessary for the agent policy we do/did implement in 8.4: https://github.com/elastic/elastic-agent/issues/1907

**Comment by blakerouse** [comment]
The difference between 8.4 and 8.6 is the idea that Elastic Agent knew what a beats configuration was, where as the 8.6 Elastic Agent should not know. Thats is what we should always keep in mind when we add something on the Elastic Agent side.

**Pull request merged**